data/model/gc.py

```diff
@@ -95,6 +118,28 @@ def purge_repository(repo, force=False):
     return True
 
 
+def _chunk_delete_all(repo, model, force=False, chunk_size=500):
+    """ Deletes all rows referencing the given repository in the given model. """
+    assert repo.state == RepositoryState.MARKED_FOR_DELETION or force
+
+    while True:
+        min_id = model.select(fn.Min(model.id)).where(model.repository == repo).scalar()
+        if min_id is None:
+            return
+
+        max_id = (
+            model.select(fn.Max(model.id))
+            .where(model.repository == repo, model.id <= min_id + chunk_size)
```

Comment:
(nitpick) It might be easier to read, and ensure the correct logic, by grouping these in parenthesis:
```
min_id + chunk_size
```
e.g.
```
.where(model.repository == repo, model.id <= (min_id + chunk_size))
```

Reply:
sure. I can change that